### ISSUE TITLE
Test Smell: it is not a good practice to write a return statement in the test case

### ISSUE THREAD

**Issue by TestSmell** [testng-team/testng#2797]
Hi!
I notice that you write a test case with a return statement.
For example, 
<img width="192" alt="image" src="https://user-images.githubusercontent.com/110402847/187157632-d2ddb266-f823-4f52-91ab-76c16c658758.png">

It is not a good idea to write the return statement in your test case based on the definition of the test case.
So I would like to know if is it really a useful test case? And no assertion is in this test case.
Maybe I think you should remove this test case from your test code. Thanks.

**Comment by juherr** [comment]
Right. It may be one of the first tests when the current best practices didn't exist.

The cannot be removed because I think there is not yet a warning against this practice.

You can propose a pull request that adds a warning message.
You can add a "for removal" tag to related tests.

**Comment by juherr** [comment]
@rajasu8 sure. Go ahead 👍

**Comment by IsaakKrut** [comment]
Hi @juherr,

I would like to work on this. Do you know how I could test the changes? When I update a test method signature and the code to return something I am getting an error "> No tests found for given includes: [NoPackageTest.test](--tests filter)". Wondering if this has been fixed already and if the tests with a return type are not getting picked up by the filter

**Comment by juherr** [comment]
@IsaakKrut 
I am uncertain if you will be able to test the modification.

The desired outcome is for a warning message to be displayed if the test methods contain a return statement. Furthermore, it should be possible to tag tests by a specific group or fix all the tests that currently have a return statement in our test suite.

To validate that the intended function is working properly, only one test case should have an explicit return value.

While I am not familiar with your testing methods, this is expected to function correctly when the test suite is executed using Gradle.

**Comment by juherr** [comment]
AI-assisted triage: **resolved** on the current codebase.

The follow-up discussion defined the useful outcome as warning when a `@Test` method has a return value rather than blindly removing such tests. Current `AnnotationHelper` now does exactly that: unless the suite explicitly enables `allow-return-values`, a non-void `@Test` is ignored with a diagnostic explaining that it has a return value and pointing to `<suite allow-return-values="true">`.

The apparently unusual internal tests are also intentional coverage: `ReturnValueTest` exercises both the default rejection and the explicit `setAllowReturnValues(true)` behavior. Removing that coverage would therefore be counterproductive.

I found no remaining implementation gap corresponding to the outcome agreed in this thread. The historical reproducer was not executed during this triage.
